openmp/libompd/gdb-plugin/ompd/ompd_handles.py

```diff
@@ -141,8 +141,7 @@ def __init__(self, thread_handle):
 	def get_current_parallel_handle(self):
 		"""Obtains the parallel handle for the parallel region associated with
 		the given thread handle."""
-		if not self.parallel_handle:
-			self.parallel_handle = ompdModule.call_ompd_get_curr_parallel_handle(self.thread_handle)
+		self.parallel_handle = ompdModule.call_ompd_get_curr_parallel_handle(self.thread_handle)
```

Comment:
```suggestion
		#TODO: invalidate thread objects based on `gdb.event.cont`. This should invalidate all internal state.
		self.parallel_handle = ompdModule.call_ompd_get_curr_parallel_handle(self.thread_handle)
```
Should work for now. Perspectively, I'd prefer proper caching of collected information.

Reply:
Done. Will take up fixing this to invalidate the thread objects in a future patch.